vm/src/org/mozartoz/truffle/nodes/control/ForNode.java

```diff
@@ -33,6 +33,7 @@ public ForNode(OzNode fromNode, OzNode toNode, OzNode procNode) {
 	public Object execute(VirtualFrame frame) {
 		long from = (long) fromDerefNode.executeDeref(fromNode.execute(frame));
 		long to = (long) toDerefNode.executeDeref(toNode.execute(frame));
+		OzProc proc = (OzProc) procNode.execute(frame);
```

Comment:
Indeed, I was wondering why I executed it inside the loop (doesn't matter once compiled but it probably does in the interpreter).
This relies on the proc being always the same but of course that's correct for a literal `for` statement.